apis/exoscale/v1/opensearch_types.go

```diff
@@ -0,0 +1,100 @@
+package v1
+
+import (
+	"reflect"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// OpenSearchParameters are the configurable fields of a OpenSearch instance.
+type OpenSearchParameters struct {
+	Maintenance     MaintenanceSpec `json:"maintenance,omitempty"`
+	Backup          BackupSpec      `json:"backup,omitempty"`
+	DBaaSParameters `json:",inline"`
+	// +kubebuilder:validation:Required
+	// Zone is the datacenter identifier in which the instance runs in.
+	Zone               Zone                 `json:"zone"`
+	Version            string               `json:"version,omitempty"`
```

Comment:
I'd add a comment about the version here (that it's only the major version most importantly).  Whatever is added as a comment for these fields will end up in the CRD as a description